shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h

```diff
@@ -122,6 +152,33 @@ class FlutterPlatformViewsController {
   void EnsureGLOverlayInitialized(int64_t overlay_id,
                                   std::shared_ptr<IOSGLContext> gl_context,
                                   GrContext* gr_context);
+  // Traverse the `mutators_stack` and return the number of clip operations.
+  int GetNumberOfClips(const MutatorsStack& mutators_stack);
+
+  // Removes extra views or add more `ChildClippingView` to ensure there are `number_of_clips`
+  // numbers of UIViews between the `child` and the `parent` including the `parent`. Returns the new
+  // parent after reconsturcting. If the returning parent is not the same as child, it will be an
+  // instance of `ChildClippingView`.
+  UIView* ReconstructClipViewsChain(int number_of_clips, UIView* child, UIView* parent);
+
+  // Apply mutators in the mutators_stack to the UIView chain that was constructed by
+  // `ReconstructClipViewsChain`
+  //
+  // Clips are applied to the super view with a CALayer mask. Transforms are applied to the current
+  // view that's at the head of the chain. For example the following mutators stack [T_1, C_2, T_3,
+  // T_4, C_5, T_6] where T denotes a transform and C denotes a clip, will result in the following
+  // UIView tree:
+  //
+  // C_2 -> C_5 -> PLATFORM_VIEW
+  // (PLATFORM_VIEW is a subview of C_5 which is a subview of C_2)
+  //
+  // T_1 is applied to C_2, T_3 and T_4 are applied to C_5, and T_6 is applied to PLATFORM_VIEW.
+  //
+  // After each clip operation, we update the head to the super view of the current head.
+  void ApplyMutators(const MutatorsStack& mutators_stack, UIView* embedded_view);
+
+  // Composite the platform view with the passed in `params`.
```

Comment:
this comment doesn't add much information on top of the method and argument names.